the-blue-alliance-ios/View Controllers/Teams/TeamSummaryTableViewController.swift

```diff
@@ -1,16 +1,320 @@
+                    return
+                }
+
+                let backgroundEvent = backgroundContext.object(with: self.event.objectID) as! Event
+                let match = Match.insert(with: modelMatch, for: backgroundEvent, in: backgroundContext)
+                if !backgroundContext.saveOrRollback() {
+                    self.showErrorAlert(with: "Unable to fetch match \(key) - database error")
+                }
+                completion(match)
+            })
+        })
+    }
+    
+    func updateSummaryInfo() {
+        summaryRows = []
+        summaryValues = []
+
+        // Rank
+        if let rank = eventStatus?.qual?.ranking?.rank {
+            summaryRows.append(TeamSummaryRow.rank)
+            summaryValues.append(rank)
+        }
+        
+        // Awards
+        // TODO: Awards aren't actually downloaded at this point... maybe kickoff a refresh?
+        let awards = event?.awards?.filtered(using: NSPredicate(format: "event == %@ AND (ANY recipients.team == %@)", event!, team!))
+        if let awards = awards, awards.count > 0 {
+            summaryRows.append(TeamSummaryRow.awards)
+            summaryValues.append(awards.count)
+        }
+        
+        // TODO: Add support for Pits
+        // https://github.com/the-blue-alliance/the-blue-alliance-ios/issues/163
+        
+        // Record
+        if let record = eventStatus?.qual?.ranking?.record, let event = event, event.year != 2015 {
+            summaryRows.append(TeamSummaryRow.record)
+            summaryValues.append(record)
+        }
+        
+        // Alliance
+        if let allianceStatusString = eventStatus?.allianceStatus {
+            summaryRows.append(TeamSummaryRow.alliance)
+            summaryValues.append(allianceStatusString)
+        }
+        
+        // Team Status
+        if let overallStatusString = eventStatus?.overallStatus {
+            summaryRows.append(TeamSummaryRow.status)
+            summaryValues.append(overallStatusString)
+        }
+        
+        // Breakdown
+        if let breakdown = eventStatus?.qual?.ranking?.infoString {
+            summaryRows.append(TeamSummaryRow.breakdown)
+            summaryValues.append(breakdown)
+        }
+
+        // Only show next/last match if the event is happening now
+        // TODO: Change this back Zach
+        if !event.isHappeningNow {
+            for (type, key) in [(TeamSummaryRow.nextMatch, eventStatus?.nextMatchKey), (TeamSummaryRow.lastMatch, eventStatus?.lastMatchKey)] {
+                guard let key = key else {
+                    continue
+                }
+                summaryRows.append(type)
+                let match = Match.findOrFetch(in: persistentContainer.viewContext, matching: NSPredicate(format: "%K == %@", #keyPath(Match.key), key))
+                if let match = match {
+                    summaryValues.append(match)
+                } else {
+                    summaryValues.append(key)
```

Comment:
Probalby kickoff our match fetches here, if we're not currently refreshing? A pretty bad place to kickoff the event but...